lib/rack/request-id-passthrough.rb

```diff
@@ -32,12 +32,25 @@ def call(env)
 
     def determine_request_id(env)
       request_id = SecureRandom.uuid
+      matches = {}
+      env.each do |key,value|
+        @headers.reverse_each do |header_name|
```

Comment:
You can use `find` instead of `reverse_each` - it will iterate in order until it finds a match, then stops. No need to continually overwrite with a higher precedence one going through them all.

```rb
env.each do |k, v|
  @headers.find do |header|
    matches[header_name] = value if same_header?(header_name, key)
  end
end
```